File: 1975_maximum_matrix_sum.java
class Solution {
    public long maxMatrixSum(int[][] a) {
        long t = 0, mn = Long.MAX_VALUE;
        int n = 0;
        for (int[] x : a) {
            for (int v : x) {
                t += Math.abs(v);
                mn = Math.min(mn, Math.abs((long) v));
                n += v < 0 ? 1 : 0;
            }
        }
        return n % 2 == 0 ? t : t - 2 * mn;
    }
}

File: Leetcode_POTD_2026/01_January_2026/1975_Maximum_Matrix_Sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long net_sum = 0;
        int min_val = INT_MAX, neg_qty = 0;
        for (auto& row_vec : matrix) {
            for (int cell_val : row_vec) {
                net_sum += abs(cell_val);
                if (cell_val < 0) neg_qty++;
                min_val = min(min_val, abs(cell_val));
            }
        }
        return (neg_qty % 2 == 0) ? net_sum : net_sum - 2LL * min_val;
    }
};
